pilot/pkg/model/xds_cache.go

```diff
@@ -74,12 +108,8 @@ func (c *inMemoryCache) Add(entry XdsCacheEntry, value *any.Any) {
 	defer c.mu.Unlock()
 	k := entry.Key()
 	c.store[k] = value
```

Comment:
Get rid of the locks because the lru cache itself has locks. These locks add up latency. Alternatively, is there an off the shelf lru cache implementation that does not use locks? We can design a simple one using hash + doubly linked list can't we?

Reply:
The locks just need to be moved around the config index

Reply:
Ah, saw your comment about simple-lru. Using that so we don't have two sets of locks. Using the 2q impl would still need locks for the config index.